File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s081988820_commented.py
import sys

def main():
    data = list(map(int, sys.stdin.read().split()))
    idx = 0
    H = data[idx]; idx += 1
    W = data[idx]; idx += 1
    c = [[0]*10 for _ in range(10)]
    for i in range(10):
        for j in range(10):
            c[i][j] = data[idx]
            idx += 1
    min_dist = [0] * 10
    for i in range(10):
        min_dist[i] = c[i][1]
    for _ in range(10):
        for i in range(10):
            for j in range(10):
                if min_dist[i] > c[i][j] + min_dist[j]:
                    min_dist[i] = c[i][j] + min_dist[j]
    ans = 0
    for _ in range(H):
        for _ in range(W):
            A = data[idx]
            idx += 1
            if A >= 0:
                ans += min_dist[A]
    print(ans)

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/C++/s081988820_commented.cpp
#include <iostream>
using namespace std;

int main() {
    int H, W;
    cin >> H >> W;
    int c[10][10];
    for (int i = 0; i < 10; i++) {
        for (int j = 0; j < 10; j++) {
            cin >> c[i][j];
        }
    }
    
    int min_dist[10];
    for (int i = 0; i < 10; i++) {
        min_dist[i] = c[i][1];
    }
    
    for (int tc = 0; tc < 10; tc++) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (min_dist[i] > c[i][j] + min_dist[j]) {
                    min_dist[i] = c[i][j] + min_dist[j];
                }
            }
        }
    }
    
    int ans = 0;
    for (int h = 0; h < H; h++) {
        for (int w = 0; w < W; w++) {
            int A;
            cin >> A;
            if (A >= 0) {
                ans += min_dist[A];
            }
        }
    }
    cout << ans << endl;
    return 0;
}
// End of Code
